src/search/landmarks/landmark_factory_merged.cc

```diff
@@ -58,10 +59,20 @@ void LandmarkFactoryMerged::generate_landmarks(
             if (!node.conjunctive && !node.disjunctive && !lm_graph->landmark_exists(lm_fact)) {
                 LandmarkNode &new_node = lm_graph->landmark_add_simple(lm_fact);
                 new_node.in_goal = node.in_goal;
+                new_node.possible_achievers.insert(
+                    node.possible_achievers.begin(), node.possible_achievers.end());
+                new_node.first_achievers.insert(
+                    node.first_achievers.begin(), node.first_achievers.end());
+                new_node.is_derived = node.is_derived;
             }
         }
     }
 
+    // TODO: It seems that disjunctive landmarks are only added if none of the
+    //  facts it is made of is also there as a simple landmark. This should
+    //  either be more general (add only if none of its subset is already there)
+    //  or it should be done only upon request (e.g., heuristics that consider
+    //  orders might want to keep all landmarks).
```

Comment:
coding conventions: we don't use `//` for many-line comments.

Regarding the comment, this probably needs more discussion, but the proposed full subset test is usually too expensive to be generally affordable. I would prefer if the comment were written to indicate that this needs further discussion, but without implying that we've already made a decision that the current behavior is not what we want.